src/applications/public-outreach-materials/libraries/library-filters.js

```diff
@@ -0,0 +1,133 @@
+export function libraryFilters(el) {
+  const typeItem = document.getElementById('outreach-type');
+  function pageNumbers() {
+    const currentPage = sessionStorage.getItem('pageNum');
+    const cards = document.querySelectorAll('.asset-card');
+    const pages = Math.ceil(cards.length / 10);
+    let activePage;
+
+    if (el.srcElement.id === 'pager-next-click') {
+      activePage = parseInt(currentPage, 10);
+      sessionStorage.setItem('pageNum', activePage++);
+    }
+    if (el.srcElement.id === 'pager-previous-click') {
+      activePage = parseInt(currentPage, 10);
+      sessionStorage.setItem('pageNum', activePage - 1);
+    }
+    if (el.srcElement.id === 'first-click') {
+      activePage = 1;
+      sessionStorage.setItem('pageNum', 1);
+    }
+    if (el.srcElement.id === 'last-click') {
+      activePage = pages;
+      sessionStorage.setItem('pageNum', activePage);
+    }
+
+    cards.forEach(element => {
+      const numVal = element.getAttribute('data-number');
+      if (numVal > activePage * 10 || numVal < activePage * 10 - 9) {
+        element.classList.add('pager-hide');
+      } else {
+        element.classList.remove('pager-hide');
+      }
+      if (activePage === undefined) {
+        if (numVal > 10) {
+          element.classList.add('pager-hide');
+        }
+      }
+    });
+
+    if (document.getElementById('va-pagination-active-num')) {
+      document.getElementById('va-pagination-active-num').innerText =
+        activePage === undefined ? 1 : activePage;
+    }
+    if (document.getElementById('total-pages')) {
+      document.getElementById('total-pages').innerText = cards.length;
+    }
+  }
+
+  sessionStorage.setItem('pageNum', 1);
+  pageNumbers();
+
+  const next = document.getElementById('pager-next-click');
```

Comment:
Since you're attaching the same listener to several ids, and then checking the id later to make sure you're taking the right action, you could instead add one listener to the element with the `va-pagination` class and then do `el.target.id` instead of `el.srcElement.id` in `pageNumbers`.

Reply:
```js
const pagingEl = document.querySelector('.va-pagination');

if (pagingEl) {
  pagingEl.addEventListener(libraryFilters);
}
```

The above would replace lines 52-70.

Then elsewhere in libraryFilters where you have this code:

```js
if (el.srcElement.id === 'pager-next-click') {
```

You would do:

```js
if (el.target.id === 'pager-next-click') {
```

Reply:
Changed per our slack convo, @jbalboni :+1: